src/lib.rs

```diff
@@ -74,42 +84,93 @@ pub trait InitWith<T> {
     /// assert_eq!(src, dest);
     /// ```
     fn init_with<F>(init: F) -> Self
-        where F: FnMut() -> T,
-              Self: Sized;
+    where
+        F: FnMut() -> T,
+        Self: Sized;
```

Comment:
What's up with adding the `Self: Sized` here?

Reply:
oh wait, never mind, i already had `Self: Sized`, carry on `>_>`

Reply:
Yeah, you need it to return `Self`, I just put it everywhere for consistency. Or because I tried to get rid of it and got confused about where I needed to put it back, either/or.

Reply:
Yeah, i read the diff wrong and thought i didn't have it before. Sorry about that.